src/components/common/ColonyActionsTable/FiltersContext/FiltersContext.tsx

```diff
@@ -0,0 +1,259 @@
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const isChecked = event.target.checked;
+      const name = event.target.name as ActivityDecisionMethod;
+
+      if (isChecked) {
+        setDecisionMethod(name);
+      } else {
+        setDecisionMethod(undefined);
+      }
+    },
+    [],
+  );
+
+  const handleDateFilterChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const isChecked = event.target.checked;
+      const name = event.target.name as keyof DateOptions;
+
+      if (isChecked) {
+        setDateFilters({ ...dateFilters, [name]: true });
+      } else {
+        setDateFilters({ ...dateFilters, [name]: false });
+      }
+    },
+    [dateFilters],
+  );
+
+  const handleCustomDateFilterChange = useCallback(
+    (date: [Date | null, Date | null]) => {
+      const [newStartDate, newEndDate] = date;
+
+      if (newStartDate && newEndDate) {
+        setDateFilters({
+          ...dateFilters,
+          custom:
+            !newStartDate && !newEndDate
+              ? undefined
+              : [newStartDate.toString(), newEndDate.toString()],
+        });
+      }
+    },
+    [dateFilters],
+  );
+
+  const activeFilters: ActivityFeedFilters = useMemo(() => {
+    const date = getDateFilter(dateFilters);
+    const actionTypes = actionTypesFilters.reduce<AnyActionType[]>(
+      (result, actionType) => {
+        const apiActionTypes = ACTION_TYPE_TO_API_ACTION_TYPES_MAP[actionType];
+
+        if (!apiActionTypes) {
+          return result;
+        }
+
+        return [...result, ...apiActionTypes];
+      },
+      [],
+    );
+
+    return {
+      ...(motionStates.length ? { motionStates } : {}),
+      ...(actionTypes.length ? { actionTypes } : {}),
+      ...(date || {}),
+      ...(decisionMethod
+        ? {
+            decisionMethod,
+          }
+        : {}),
+      ...(searchFilter ? { search: searchFilter } : {}),
+    };
+  }, [
+    actionTypesFilters,
+    dateFilters,
+    decisionMethod,
+    motionStates,
+    searchFilter,
+  ]);
+
+  const handleResetFilters = useCallback((category: string) => {
+    switch (category) {
+      case formatText({ id: 'activityFeedTable.filters.actionType' }): {
+        return setActionTypesFilters([]);
+      }
+      case formatText({ id: 'activityFeedTable.filters.status' }): {
+        return setMotionStates([]);
+      }
+      case formatText({ id: 'activityFeedTable.filters.decisionMethod' }): {
+        return setDecisionMethod(undefined);
+      }
+      case formatText({ id: 'activityFeedTable.filters.date' }):
+      case formatText({ id: 'activityFeedTable.filters.date.custom' }): {
+        return setDateFilters({
+          pastHour: false,
+          pastDay: false,
+          pastWeek: false,
+          pastMonth: false,
+          pastYear: false,
+          custom: undefined,
+        });
+      }
+      default: {
+        return undefined;
+      }
+    }
+  }, []);
```

Comment:
Just coming back to this from #1927, there should be an enum/union type identifying those filters instead of relying on i18n strings

Also the name `category` is a bit unclear, it's the field you're filtering on, right?